src/EnergyPlus/ManageElectricPower.cc

```diff
@@ -3192,6 +3175,82 @@ namespace ManageElectricPower {
 
 	//*****************************************************************************************************************
 
+	bool 
+	determineCurrentForBatteryDischarge( 
+		Real64& curI0, 
+		Real64& curT0, 
+		Real64& curVolt,
+		Real64 const Pw, 
+		Real64 const q0,
+		int const CurveNum,
+		Real64 const k,
+		Real64 const c,
+		Real64 const qmax,
+		Real64 const E0c,
+		Real64 const InternalR )
+	{
+		// FUNCTION INFORMATION:
+		//       AUTHOR         B. Griffith
+		//       DATE WRITTEN   June-August 2008
+		//       MODIFIED       BG May 2009, added EMS
+		//                      BN (FSEC) Feb 2010 (pass out two storage values)
+		//                      Y. KyungTae & W. Wang July-August, 2011 Added a battery model
+		//       RE-ENGINEERED  Jason Glazer, GARD Analytics, February 2015, refactor charge calculation into a function
+
+		// PURPOSE OF THIS FUNCTION:
+		// Calculate the current for battery discharge in a separate function so that it could be called from the unit tests
+
```

Comment:
@JasonGlazer I added this function information block - maybe it isn't necessary but I think it is helpful. Does it look OK to you?